File: Python/Decorator/MobilePhone.py
# -*- coding:utf-8 -*-

class MobilePhone:

    def __init__(self, str):
        self.phoneName = str

    def ShowDecorate(self):
        print "%s%s"%(self.phoneName,"的装饰")

class Iphone(MobilePhone):

    def __init__(self):
        MobilePhone.__init__(self, "Iphone")

class Samsung(MobilePhone):

    def __init__(self):
        MobilePhone.__init__(self, "Samsung")

File: CPlusPlus/Decorator/src/MobilePhone.cpp
/*************************************************************************
    > File Name: MobilePhone.cpp
    > Author: ma6174
    > Mail: ma6174@163.com 
    > Created Time: 2017年03月02日 星期四 14时10分44秒
 ************************************************************************/

#include"MobilePhone.h"
#include<string>
#include<iostream>

using namespace std;

MobilePhone::MobilePhone(const string& str):phoneName(str)
{
	;
}

MobilePhone::~MobilePhone()
{
	;
}

void MobilePhone::ShowDecorate()
{
	cout<<phoneName<<"的装饰"<<endl;
}

string MobilePhone::GetPhoneName()
{
	return phoneName;
}
